client/app/bundles/course/material/folders/components/buttons/KnowledgeBaseSwitch.tsx

```diff
@@ -0,0 +1,107 @@
+// currently not in use
+import { FC, useEffect, useState } from 'react';
+import { defineMessages } from 'react-intl';
+import { Switch } from '@mui/material';
+import { MaterialWorkflowState } from 'types/course/material/folders';
+
+import { useAppDispatch } from 'lib/hooks/store';
+import toast from 'lib/hooks/toast';
+import useTranslation from 'lib/hooks/useTranslation';
+
+import { chunkMaterial, removeChunks } from '../../operations';
+
+interface Props {
+  currFolderId: number;
+  itemId: number;
+  itemName: string;
+  isConcrete: boolean;
+  canEdit: boolean;
+  state: MaterialWorkflowState;
+  type: 'subfolder' | 'material';
+}
+
+const translations = defineMessages({
+  addSuccess: {
+    id: 'course.material.folders.WorkbinTableButtons.addFailure',
+    defaultMessage: ' has been added to knowledge base',
+  },
+  addFailure: {
+    id: 'course.material.folders.WorkbinTableButtons.addFailure',
+    defaultMessage: ' could not be added to knowledge base',
+  },
+  removeSuccess: {
+    id: 'course.material.folders.WorkbinTableButtons.removeSuccess',
+    defaultMessage: ' has been removed from knowledge base',
+  },
+  removeFailure: {
+    id: 'course.material.folders.WorkbinTableButtons.removeFailure',
+    defaultMessage: ' could not be removed from knowledge base',
+  },
+});
+
+const KnowledgeBaseSwitch: FC<Props> = (props) => {
+  const { currFolderId, itemId, itemName, isConcrete, canEdit, state, type } =
+    props;
+  const { t } = useTranslation();
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useAppDispatch();
+  const onAdd = (): void => {
+    setIsLoading(true);
+    dispatch(
+      chunkMaterial(
+        currFolderId,
+        itemId,
+        () => {
+          setIsLoading(false);
+          toast.success(`"${itemName}" ${t(translations.addSuccess)}`);
```

Comment:
We don't interpolate like this.  
This will cause issues when we change languages where the Subject-Verb-Object ordering is different.

search for other examples, e.g. `t(translations.moveTabsToCategoryThenDelete, {category: "some title"})`

Reply:
updated